hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/SimpleKeyGenerator.java

```diff
@@ -49,7 +50,9 @@ public SimpleKeyGenerator(TypedProperties props) {
   SimpleKeyGenerator(TypedProperties props, String recordKeyField, String partitionPathField) {
     super(props);
     // Make sure key-generator is configured properly
-    validateRecordKey(recordKeyField);
+    if (!doAutoGenerateRecordKeys()) {
+      validateRecordKey(recordKeyField);
+    }
```

Comment:
move it inside `validateRecordKey`? that method should be self-contained, return true if autoGenerateRecordKeys is enabled.